crates/codegen/schema/src/validation/ast/visitors.rs

```diff
@@ -243,22 +252,25 @@ impl Receiver for ParserRef {
 
 impl Receiver for ParserDefinition {
     fn receive<V: Visitor>(&self, visitor: &mut V, reporter: &mut Reporter) {
-        match &self {
-            ParserDefinition::Choice(expressions) | ParserDefinition::Sequence(expressions) => {
-                for expression in expressions {
+        if visitor.visit_parser_definition(self, reporter) == VisitorResponse::StepIn {
```

Comment:
To avoid subtle bugs [like this](https://github.com/NomicFoundation/slang/pull/332#discussion_r1130506990), I suggest only implementing `Receiver` for types that have `Visitor::visit_**` visitors. For example, `ParserRef`, `ScannerRef`, etc... but not `ParserDefinition` or `ScannerDefinition`, etc....
This ensures that each reciever correctly checks `if visitor.visit(self) == Response::StepIn` before actually stepping in.

Reply:
As it now is.